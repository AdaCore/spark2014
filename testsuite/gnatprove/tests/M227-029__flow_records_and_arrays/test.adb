package body Test is 

   type Coordinate is record
      X : Integer;
      Y : Integer;
      Z : Integer;
   end record;

   Origin : constant Coordinate := Coordinate'(others => 0);

   type Ammo_Type is (Bullets, Nails, Rockets, Power);

   type Ammo_Store is array (Ammo_Type) of Natural;

   type Player is record
      Position   : Coordinate;
      Look_Yaw   : Integer;
      Look_Pitch : Integer;
      Ammo       : Ammo_Store;
   end record;

   type Player_Id is range 1 .. 8;

   type Players is array (Player_Id range <>) of Player;

   function New_Player_01 return Player is
      P : Player;
   begin
      P := Player'(Position   => (0, 0, 0),
                   Look_Yaw   => 0,
                   Look_Pitch => 0,
                   Ammo       => (others => 0));
      return P;
   end New_Player_01;

   function New_Player_02 return Player is
      P : Player;
   begin
      P.Position   := (0, 0, 0);
      P.Look_Yaw   := 0;
      P.Look_Pitch := 0;
      P.Ammo       := (others => 0);
      return P;
   end New_Player_02;

   function New_Player_03 return Player is
      P : Player;
   begin
      P.Position.X := 0;
      P.Position.Y := 0;
      P.Position.Z := 0;
      P.Look_Yaw   := 0;
      P.Look_Pitch := 0;
      P.Ammo       := (others => 0);
      return P;
   end New_Player_03;

end Test;
