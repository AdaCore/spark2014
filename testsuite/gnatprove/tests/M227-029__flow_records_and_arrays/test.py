from test_support import *
from glob import glob

do_flow(opt=sorted(glob("*.adb")))
