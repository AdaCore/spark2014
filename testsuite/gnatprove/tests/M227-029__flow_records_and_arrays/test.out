message_tests.adb:40:33: warning: unused initial value of "C"
message_tests.adb:40:33: warning: unused variable "C"
message_tests.adb:48:32: "C.Flags.A" is not initialized in "Not_All_Used_Out"
message_tests.adb:48:32: "C.Flags.B" is not initialized in "Not_All_Used_Out"
message_tests.adb:48:32: "C.W" is not initialized in "Not_All_Used_Out"
message_tests.adb:66:15: "F.B" is not initialized
parameter_tests.adb:36:32: "G.Y" is not initialized
parameter_tests.adb:50:33: warning: unused initial value of "G"
parameter_tests.adb:50:36: warning: "H" is not modified, could be "input"
   compilation of message_tests.adb failed
   compilation of parameter_tests.adb failed


gnatprove: error during translation to intermediate language, aborting.
gprbuild: *** compilation phase failed
